presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergHiveMetadata.java

```diff
@@ -371,6 +401,147 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+    {
+        Map<String, String> properties = ImmutableMap.<String, String>builder()
+                .put(TABLE_COMMENT, "Presto View")
+                .put(PRESTO_VIEW_FLAG, "true")
+                .put(PRESTO_VERSION_NAME, prestoVersion)
+                .put(PRESTO_QUERY_ID_NAME, session.getQueryId())
+                .put(TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE)
+                .put(METADATA_LOCATION_PROP, "")
+                .build();
+
+        List<Column> columns = new ArrayList<>();
+        MetastoreContext metastoreContext = getMetastoreContext(session);
+        ColumnConverter columnConverter = metastoreContext.getColumnConverter();
+
+        for (ColumnMetadata column : viewMetadata.getColumns()) {
+            try {
+                HiveType hiveType = toHiveType(new HiveTypeTranslator(), column.getType());
+                columns.add(new Column(column.getName(), hiveType, Optional.ofNullable(column.getComment()), columnConverter.getTypeMetadata(hiveType, column.getType().getTypeSignature())));
+            }
+            catch (PrestoException e) {
+                // if a view uses any unsupported hive types, include only a dummy column value
+                if (e.getErrorCode().equals(NOT_SUPPORTED.toErrorCode())) {
+                    columns = ImmutableList.of(
+                            new Column(
+                                    "dummy",
+                                    HIVE_STRING,
+                                    Optional.of(format("Using dummy because column %s uses unsupported Hive type %s ", column.getName(), column.getType())),
+                                    Optional.empty()));
+                    break;
+                }
+                else {
+                    throw e;
+                }
+            }
+        }
+
+        SchemaTableName viewName = viewMetadata.getTable();
+
+        Table.Builder tableBuilder = Table.builder()
+                .setDatabaseName(viewName.getSchemaName())
+                .setTableName(viewName.getTableName())
+                .setOwner(session.getUser())
+                .setTableType(VIRTUAL_VIEW)
+                .setDataColumns(columns)
+                .setPartitionColumns(ImmutableList.of())
+                .setParameters(properties)
+                .setViewOriginalText(Optional.of(encodeViewData(viewData)))
+                .setViewExpandedText(Optional.of("/* Presto View */"));
+
+        tableBuilder.getStorageBuilder()
+                .setStorageFormat(VIEW_STORAGE_FORMAT)
+                .setLocation("");
+        Table table = tableBuilder.build();
+        PrincipalPrivileges principalPrivileges = buildInitialPrivilegeSet(session.getUser());
+
+        Optional<Table> existing = metastore.getTable(metastoreContext, viewName.getSchemaName(), viewName.getTableName());
+        if (existing.isPresent()) {
+            if (!replace || !MetastoreUtil.isPrestoView(existing.get())) {
```

Comment:
Import static method `isPrestoView` and use that here

Reply:
Done.